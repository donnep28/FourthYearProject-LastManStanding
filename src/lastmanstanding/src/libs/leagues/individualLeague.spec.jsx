import React from 'react';
import {shallow} from 'enzyme';
import IndividualLeague from './individualLeague';

describe('Individual League', () => {
  beforeAll(() => {
    jest.resetAllMocks();
  });

  const props = {
    username: "mockUsername",
    sub: "mockSub",
    leagueId: "mockLeagueId",
    individualLeague: true,
    setIndividualLeague: jest.fn(),
    fixtures: {
      data: [
        [],
        [
          {
            'AwayScore': '0',
            'AwayTeam': 'Arsenal FC',
            'AwayTeamCrest': 'https://crests.football-data.org/57.svg',
            'createdTime': '2021-02-11 12:40:00.561002',
            'GameWeek': '24',
            'HomeScore': '1',
            'HomeTeam': 'Aston Villa FC',
            'HomeTeamCrest': 'https://crests.football-data.org/58.svg',
            'MatchID': 'Aston Villa FC-Arsenal FC',
            'Winner': 'Aston Villa FC',
          },
          {
            'AwayScore': '1',
            'AwayTeam': 'Brighton FC',
            'AwayTeamCrest': 'https://crests.football-data.org/397.svg',
            'createdTime': '2021-02-11 12:40:00.654459',
            'GameWeek': '24',
            'HomeScore': '1',
            'HomeTeam': 'Burnley FC',
            'HomeTeamCrest': 'https://crests.football-data.org/328.svg',
            'MatchID': 'Burnley FC-Brighton FC',
            'Winner': 'Draw',
          },
          {
            'AwayScore': '0',
            'AwayTeam': 'West Ham United FC',
            'AwayTeamCrest': 'https://crests.football-data.org/563.svg',
            'createdTime': '2021-02-11 12:40:00.694246',
            'GameWeek': '24',
            'HomeScore': '0',
            'HomeTeam': 'Fulham FC',
            'HomeTeamCrest': 'https://crests.football-data.org/63.svg',
            'MatchID': 'Fulham FC-West Ham United FC',
            'Winner': 'Draw',
          },
        ],
        [
            {
            'AwayTeam': 'Leeds United FC',
            'AwayTeamCrest': 'https://crests.football-data.org/341.svg',
            'createdTime': '2021-02-11 12:40:01.254269',
            'FixtureID': 'Arsenal FC-Leeds United FC',
            'GameWeek': '24',
            'HomeTeam': 'Arsenal FC',
            'HomeTeamCrest': 'https://crests.football-data.org/57.svg',
            },
            {
            'AwayTeam': 'Aston Villa FC',
            'AwayTeamCrest': 'https://crests.football-data.org/58.svg',
            'createdTime': '2021-02-11 12:40:01.134268',
            'FixtureID': 'Brighton FC-Aston Villa FC',
            'GameWeek': '24',
            'HomeTeam': 'Brighton FC',
            'HomeTeamCrest': 'https://crests.football-data.org/397.svg',
            },
            {
            'AwayTeam': 'Newcastle United FC',
            'AwayTeamCrest': 'https://crests.football-data.org/67.svg',
            'createdTime': '2021-02-11 12:40:01.374223',
            'FixtureID': 'Chelsea FC-Newcastle United FC',
            'GameWeek': '24',
            'HomeTeam': 'Chelsea FC',
            'HomeTeamCrest': 'https://crests.football-data.org/61.svg',
            },
        ],]
    },
  };

  describe('Snapshot', () => {
    it('renders correctly', () => {
      const leagueInfo = {data: []}
      const component = shallow(
          <IndividualLeague {...props} leagueInfo={leagueInfo}/>,
      );
      expect(component).toMatchSnapshot();
    });
  });
});