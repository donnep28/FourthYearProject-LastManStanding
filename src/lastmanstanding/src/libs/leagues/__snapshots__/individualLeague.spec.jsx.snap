// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Individual League Snapshot renders correctly 1`] = `
<div
  style={
    Object {
      "paddingTop": "30px",
    }
  }
>
  <WithStyles(ForwardRef(Card))
    style={
      Object {
        "boxShadow": "6px 8px 8px 8px rgb(162, 162, 163)",
        "padding": "12px",
      }
    }
  >
    <div
      align="right"
    >
      <WithStyles(ForwardRef(Button))
        onClick={[Function]}
      >
        <Memo(ForwardRef(CloseIcon)) />
      </WithStyles(ForwardRef(Button))>
    </div>
    <WithStyles(ForwardRef(CardContent))>
      <div
        className="leagueLoading"
      >
        <WithStyles(ForwardRef(CircularProgress))
          style={
            Object {
              "color": "#490050",
            }
          }
        />
      </div>
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(Card))>
</div>
`;
